front/app/components/admin/GraphCards/PostsByTimeCard/Chart.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useMemo } from 'react';
+
+// styling
+import { MARGINS } from 'components/admin/GraphCards/_utils/style';
+
+// components
+import LineBarChart from 'components/admin/Graphs/LineBarChart';
+
+// utils
+import { isNilOrError, NilOrError } from 'utils/helperUtils';
+import { toThreeLetterMonth } from 'utils/dateUtils';
+import { generateEmptyData } from './generateEmptyData';
+
+// typings
+import {
+  ProjectId,
+  Dates,
+  Resolution,
+  Layout,
+} from 'components/admin/GraphCards/typings';
+import { TimeSeries } from './usePostsByTime/typings';
+
+type Props = ProjectId &
+  Dates &
+  Resolution & {
+    timeSeries: TimeSeries | NilOrError;
+    innerRef: React.RefObject<any>;
+    layout?: Layout;
+  };
+
+const Chart = ({
+  timeSeries,
+  projectId,
+  startAtMoment,
+  endAtMoment,
+  resolution,
+  innerRef,
+  layout = 'wide',
+}: Props) => {
+  const emptyData = useMemo(
+    () => generateEmptyData(startAtMoment, endAtMoment, resolution),
+    [startAtMoment, endAtMoment, resolution]
+  );
+
+  const formatTick = (date: string) => {
+    return toThreeLetterMonth(date, resolution);
+  };
+
+  // Avoids unmounted component state update warning
+  if (timeSeries === undefined) {
+    return null;
+  }
+
+  const noData = isNilOrError(timeSeries) || !!projectId;
```

Comment:
Should this be `!projectId` instead of `!!projectId`?

Reply:
Right, i copied it from registrations chart, but it makes sense only there.